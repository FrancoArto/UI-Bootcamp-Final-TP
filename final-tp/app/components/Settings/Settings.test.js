import React from 'react';
import {mount, shallow} from 'enzyme'
import Settings from './Settings';



it('renders without crashing', () => {
  const inputSettings = {
    verified: false,
    following: true,
    defaultInfo: false,
    withLink: true,
    withTruncatedText: false
  }
  const rendered = shallow(<Settings settings={inputSettings}/>)
  expect(rendered).toMatchSnapshot();
});

it('handles verified setting change', () => {
  const inputSettings = {
    verified: false,
    following: true,
    defaultInfo: false,
    withLink: true,
    withTruncatedText: false
  }

  var outputSettings = inputSettings

  function onVerifiedChange () {
    outputSettings.verified = !inputSettings.verified
  } 
  const wrapper = mount(<Settings 
    settings={inputSettings}
    onVerifiedChange={onVerifiedChange}/>)

  const checkbox = wrapper.find('CheckBox').first();
  checkbox.props().onPress();
  wrapper.setProps({
    settings: outputSettings
  })
  wrapper.update()
  expect(wrapper.props().settings).toEqual(outputSettings)
});

it('handles following setting change', () => {
  const inputSettings = {
    verified: false,
    following: true,
    defaultInfo: false,
    withLink: true,
    withTruncatedText: false
  }
  
  var outputSettings = inputSettings

  function onFollowingChange () {
    outputSettings.following = !inputSettings.following
  } 
  const wrapper = mount(<Settings 
    settings={inputSettings}
    onFollowingChange={onFollowingChange}/>)

  const checkbox = wrapper.find('CheckBox').at(1);
  checkbox.props().onPress();
  wrapper.setProps({
    settings: outputSettings
  })
  wrapper.update()
  expect(wrapper.props().settings).toEqual(outputSettings)
});

it('handles default info setting change', () => {
  const inputSettings = {
    verified: false,
    following: true,
    defaultInfo: false,
    withLink: true,
    withTruncatedText: false
  }
  
  var outputSettings = inputSettings

  function onDefaultInfoChange () {
    outputSettings.defaultInfo = !inputSettings.defaultInfo
  } 
  const wrapper = mount(<Settings 
    settings={inputSettings}
    onDefaultInfoChange={onDefaultInfoChange}/>)

  const checkbox = wrapper.find('CheckBox').at(2);
  checkbox.props().onPress();
  wrapper.setProps({
    settings: outputSettings
  })
  wrapper.update()
  expect(wrapper.props().settings).toEqual(outputSettings)
});

it('handles with link setting change', () => {
  const inputSettings = {
    verified: false,
    following: true,
    defaultInfo: false,
    withLink: true,
    withTruncatedText: false
  }
  
  var outputSettings = inputSettings

  function onWithLinkChange () {
    outputSettings.withLink = !inputSettings.withLink
  } 
  const wrapper = mount(<Settings 
    settings={inputSettings}
    onWithLinkChange={onWithLinkChange}/>)

  const checkbox = wrapper.find('CheckBox').at(3);
  checkbox.props().onPress();
  wrapper.setProps({
    settings: outputSettings
  })
  wrapper.update()
  expect(wrapper.props().settings).toEqual(outputSettings)
});

it('handles truncated text setting change', () => {
  const inputSettings = {
    verified: false,
    following: true,
    defaultInfo: false,
    withLink: true,
    withTruncatedText: false
  }
  
  var outputSettings = inputSettings

  function onWithTruncatedTextChange () {
    outputSettings.withTruncatedText = !inputSettings.withTruncatedText
  } 
  const wrapper = mount(<Settings 
    settings={inputSettings}
    onWithTruncatedTextChange={onWithTruncatedTextChange}/>)

  const checkbox = wrapper.find('CheckBox').last();
  checkbox.props().onPress();
  wrapper.setProps({
    settings: outputSettings
  })
  wrapper.update()
  expect(wrapper.props().settings).toEqual(outputSettings)
});