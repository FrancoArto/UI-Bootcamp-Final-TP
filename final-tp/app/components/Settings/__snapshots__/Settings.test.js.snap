// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Styled(Content)>
  <Styled(Card)>
    <Styled(CardItem)
      style={
        Object {
          "backgroundColor": "#79a6d2",
        }
      }
    >
      <Styled(Body)>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFFFFF",
              "fontWeight": "800",
            }
          }
        >
          Silence notifications from...
        </Text>
      </Styled(Body)>
    </Styled(CardItem)>
  </Styled(Card)>
  <Styled(ListItem)>
    <Styled(Body)>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        People who have not verified the account
      </Text>
    </Styled(Body)>
    <Styled(CheckBox)
      checked={false}
      onPress={[Function]}
    />
  </Styled(ListItem)>
  <Styled(ListItem)>
    <Styled(Body)>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        People who do not follow
      </Text>
    </Styled(Body)>
    <Styled(CheckBox)
      checked={true}
      onPress={[Function]}
    />
  </Styled(ListItem)>
  <Styled(ListItem)>
    <Styled(Body)>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        People who have default profile information
      </Text>
    </Styled(Body)>
    <Styled(CheckBox)
      checked={false}
      onPress={[Function]}
    />
  </Styled(ListItem)>
  <Styled(ListItem)>
    <Styled(Body)>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Tweets that contains a link
      </Text>
    </Styled(Body)>
    <Styled(CheckBox)
      checked={true}
      onPress={[Function]}
    />
  </Styled(ListItem)>
  <Styled(ListItem)>
    <Styled(Body)>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Tweets that has text truncated
      </Text>
    </Styled(Body)>
    <Styled(CheckBox)
      checked={false}
      onPress={[Function]}
    />
  </Styled(ListItem)>
</Styled(Content)>
`;
